Gems/ScriptCanvas/Code/Editor/Include/ScriptCanvas/Assets/ScriptCanvasFileHandling.h

```diff
@@ -41,14 +41,20 @@ namespace ScriptCanvasEditor
         AZStd::string ToString(size_t depth = 0) const;
     };
 
-    AZ::Outcome<SourceHandle, AZStd::string> LoadFromFile(AZStd::string_view path);
-
-    AZ::Outcome<void, AZStd::string> LoadDataFromJson
+    // both success and failure return JSON deserialization errors, regardless of parsing / loading success
+    AZ::Outcome<AZStd::string, AZStd::string> LoadDataFromJson
         ( ScriptCanvas::ScriptCanvasData& dataTarget
         , AZStd::string_view source
         , AZ::SerializeContext& serializeContext);
 
     AZ::Outcome<EditorAssetTree, AZStd::string> LoadEditorAssetTree(SourceHandle handle, EditorAssetTree* parent = nullptr);
-        
+
+    struct FileLoadSuccess
+    {
+        SourceHandle handle;
+        AZStd::string deserializationErrors;
```

Comment:
It works, but looks weird. Maybe rename as FileLoadReport? Consumer will not expect errors in success object?

Reply:
I would use the word Report if the result could be a failure or a success. This is only used in the success full case. handle is expected to have a good, useable graph pointer in it. The deserializationErrors are there because the new mode of deserialization is to load the graph even if there are errors, and just keep them reportable.